### PR TITLE
[MRG] Add handedness to participant files

### PR DESCRIPTION
**Pull request by dominikwelke**
PR Description
--------------

hi all - 

here is a small PR that adds the subjects' handedness to the participant files (related to #302) 
i also added generic timezone information to some datetime-objects where it was missing (missing tz-info can caused errors).

cheers, Dominik

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"
- [ ] Commit history does not contain any merge commits

### CODE REVIEW

**Comment by dominikwelke** [comment]
`his_id` could be added quickly but wouldnt be too useful, imho
fields for subjects full name are also included in `raw.info`, but this is sensitive information

File: mne_bids/write.py [hunk 1]
```diff
@@ -300,20 +301,25 @@ def _participants_tsv(raw, subject_id, fname, overwrite=False,
             if isinstance(meas_date, datetime):
                 meas_datetime = meas_date
             else:
-                meas_datetime = datetime.fromtimestamp(meas_date)
+                meas_datetime = datetime.fromtimestamp(meas_date, tz=timezone.utc)
```
Comment (sappelhoff) [inline mne_bids/write.py]: why was adding a timezone necessary?

File: mne_bids/write.py [hunk 2]
```diff
@@ -300,20 +301,25 @@ def _participants_tsv(raw, subject_id, fname, overwrite=False,
...
             subject_age = _age_on_date(bday, meas_datetime)
         else:
             subject_age = "n/a"
 
-    data.update({'age': [subject_age], 'sex': [sex]})
+        # add handedness
+        hand_options = {0: 'n/a', 1: 'R', 2: 'L', 3: 'A'}  # not according to mne conventions (only r/l)
+        hand = hand_options[subject_info.get('hand', 0)]
```
Comment (sappelhoff) [inline mne_bids/write.py]: for readability reasons I would move this to 291 right below the sex variable declaration (with a blank line in between).

then we would have two short extractions, and then a long one.

Also: What do you mean by the comment "not according to mne conventions (only r/l)"?

File: mne_bids/write.py (hunk 1 above)
Comment (dominikwelke) [reply mne_bids/write.py]: yes, sorry for being unclear:
this is actually unrelated to the handedness information..

i saw that the datetime object for the participants birthdate has a timezone, while it was not specified for the date of recording. 
in my testing without the edit i got the following error:
```
File "/Users/dominik.welke/Work/git_contributions/mne-bids/mne_bids/write.py", line 1241, in write_raw_bids
    verbose)
  File "/Users/dominik.welke/Work/git_contributions/mne-bids/mne_bids/write.py", line 305, in _participants_tsv
    subject_age = _age_on_date(bday, meas_datetime)
  File "/Users/dominik.welke/Work/git_contributions/mne-bids/mne_bids/utils.py", line 352, in _age_on_date
    if exp_date < bday:
TypeError: can't compare offset-naive and offset-aware datetimes
```

File: mne_bids/write.py (hunk 2 above)
Comment (dominikwelke) [reply mne_bids/write.py]: i rearranged the order..

concerning the comment: 
the mne [documentation](https://mne.tools/dev/generated/mne.Info.html#mne.Info) suggests only two options: 
`hand : int`
`Handedness (1=right, 2=left).`

no n/a, no ambidextrous. 
this was more of a reminder for myself to open an issue on mne's git

Comment (sappelhoff) [reply mne_bids/write.py]: I see, in that case I would prefer

```
        # XXX: MNE currently only handles R/L, follow <insert MNE issue ID here>
        hand_options = {0: 'n/a', 1: 'R', 2: 'L', 3: 'A'}
```

and if you don't want to make an issue now, don't put <insert MNE issue ID here>

File: mne_bids/write.py [hunk 3]
```diff
@@ -283,11 +283,16 @@ def _participants_tsv(raw, subject_id, fname, overwrite=False,
...
     subject_age = "n/a"
     sex = "n/a"
+    hand = 'n/a'
     subject_info = raw.info['subject_info']
     if subject_info is not None:
         sexes = {0: 'n/a', 1: 'M', 2: 'F'}
         sex = sexes[subject_info.get('sex', 0)]
 
+        # add handedness
+        hand_options = {0: 'n/a', 1: 'R', 2: 'L', 3: 'A'}
```
Comment (jasmainak) [inline mne_bids/write.py]: did you invent these options? according to https://mne.tools/dev/generated/mne.Info.html only 1 and 2 are possible options

Comment (sappelhoff) [reply mne_bids/write.py]: see discussion here @jasmainak https://github.com/mne-tools/mne-bids/pull/354#discussion_r382652813

File: mne_bids/write.py (hunk 2 above)
Comment (dominikwelke) [reply mne_bids/write.py]: i followed your suggestion and included a link to the issue

**Comment by jasmainak** [comment]
CIs are unfortunately failing

**Comment by dominikwelke** [comment]
could you guys maybe give me a hand with circleci?


do i understand correctly, that it cant build the doc because example data do not contain handedness info?

**Comment by sappelhoff** [comment]
> do i understand correctly, that it cant build the doc because example data do not contain handedness info?

yes, this is where it fails (in _participants_tsv):

```Python
    if os.path.exists(fname):
        orig_data = _from_tsv(fname)
        # whether the new data exists identically in the previous data
        exact_included = _contains_row(orig_data,
                                       {'participant_id': subject_id,
                                        'age': subject_age,
                                        'sex': sex,
                                        'hand': hand})
```

it somehow enters the `if os.path.exists(fname):` clause, and then reads in an `orig_data` that doesn't properly exist in the format we expect.

I am just at a loss why it suddenly enters that clause. The `fname` should not exist, considering how the flow of the example is: https://github.com/mne-tools/mne-bids/blob/master/examples/convert_mne_sample.py

**Comment by dominikwelke** [comment]
fyi, when i filed the [mne-issue](https://github.com/mne-tools/fiff-constants/issues/25) to adapt add `hand`-options, someone else jumped in to also add further `sex` -options.

it shold be possible to take the current options directly from `mne`, rather then having them hard coded in `mne-bids` code. this would be more robust, but maybe also a bit of unneccessary overkill.. what do you think?

**Comment by sappelhoff** [comment]
So yes, the error of CircleCI was spurious and build-related. We can merge this :-)

> it shold be possible to take the current options directly from mne, rather then having them hard coded in mne-bids code. this would be more robust, but maybe also a bit of unneccessary overkill.. what do you think?

Once this is fully implemented in MNE I would agree that taking it from there would be superior to re-defining it in mne-bids.

But until that MNE issue has been fully resolved, we can leave the state as it is. So let's track https://github.com/mne-tools/fiff-constants/issues/25 and then move on from there.

Thanks @dominikwelke!

File: doc/whats_new.rst [hunk 4]
```diff
@@ -22,6 +22,7 @@ Current
 Changelog
 ~~~~~~~~~
 
+- Added handedness to participant files, by `Dominik Welke`_ (`#354 <https://github.com/mne-tools/mne-bids/pull/354>`_)
```
Comment (jasmainak) [inline doc/whats_new.rst]: you should add the change to the bottom of the changelog not the top.

Comment (dominikwelke) [reply doc/whats_new.rst]: ok, next time ill do that..

**Pull request merged**